Unity/TypeTree.cs

```diff
@@ -35,11 +36,26 @@ public string GetString(uint offset)
             return Marshal.PtrToStringAnsi(new IntPtr(tree.StringBuffer.Ptr + offset));
         }
 
+        public void CreateNodes()
+        {
+            tree.CreateNodes(this);
+            StringBuffer = new byte[tree.StringBuffer.Size];
+            fixed (byte* destination = StringBuffer)
+                Buffer.MemoryCopy(tree.StringBuffer.Ptr, destination, StringBuffer.Length, StringBuffer.Length);
+        }
+
+        public TypeTreeNode this[int index] => tree.Nodes[index];
+
+        public int Count => tree.Nodes.Count;
+
+        public byte[] StringBuffer;

```

Comment:
Is this supposed to be public? It should be changed to `public IReadOnlyList<byte> StringBuffer { get; private set; }` if so.

Reply:
A public string buffer is needed for binary struct dumps, shouldn't be any issue with `public IReadOnlyList<byte> StringBuffer { get; private set; }`

Reply:
Is there any reason that it doesn't just return `tree.StringBuffer` in that case?

Reply:
It might be worthwhile to implement `IList<T>` or `IReadOnlyList<T>` on `DynamicArray<T>` to make that a bit easier, or just having a wrapper type.

Reply:
DynamicArray isn't public, perhaps ITypeTreeImpl.StringBuffer should be IReadOnlyList<byte>?

Reply:
That sounds good to me

Reply:
Changing `ITypeTreeImpl.StringBuffer` to `IReadOnlyList<byte>` makes the implementation of GetString a bit awkard.

https://github.com/DaZombieKiller/TypeTreeDumper/blob/627380d48855bd0827206711f97c9512675a85ad/Unity/TypeTree.cs#L35

i think it is easier to keep ITypeTreeImpl.StringBuffer the way it is and just add a getter for TypeTree.StringBuffer.

Btw, are strings ansi or utf8? I'm guessing the question is academic and only ascii characters are used.

Reply:
> i think it is easier to keep ITypeTreeImpl.StringBuffer the way it is and just add a getter for TypeTree.StringBuffer.

Yeah, that sounds fine.

> Btw, are strings ansi or utf8? I'm guessing the question is academic and only ascii characters are used.

I _think_ they're ANSI, but it might be worthwhile to switch to [PtrToStringUTF8](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.ptrtostringutf8) just in case.